RULES.md

```diff
@@ -1,536 +1,1037 @@
+|-----------------------------------------------	|---------------------------	|-------------------------------------------------------|
+| [`UnknownColumnNotice`](#UnknownColumnNotice) 	| A column name is unknown. 	| unknown_column_notice                                 |
+| [`UnknownFileNotice`](#UnknownFileNotice)     	| A file is unknown.        	|unknown_file_notice                                    |
 
 <a name="SYSTEM_ERRORS"/>
 
 ## Table of system errors
 
-| Name                                                                    	| Description                                            	|
-|-------------------------------------------------------------------------	|--------------------------------------------------------	|
-| [`IOError`](#IOError)                                                   	| Error in IO operation.                                 	|
-| [`RuntimeExceptionInLoaderError`](#RuntimeExceptionInLoaderError)       	| RuntimeException while loading GTFS dataset in memory. 	|
-| [`RuntimeExceptionInValidatorError`](#RuntimeExceptionInValidatorError) 	| RuntimeException while validating GTFS archive.        	|
-| [`ThreadExecutionError`](#ThreadExecutionError)                         	| ExecutionException during multithreaded validation     	|
-| [`URISyntaxError`](#URISyntaxError)                                     	| A string could not be parsed as a URI reference.       	|
+| System error name                                                         | Description                                            	| Notice Code (as it appears in the `system_errors.json` file) |
+|-------------------------------------------------------------------------	|--------------------------------------------------------	|--------------------------------------------------------------|
+| [`IOError`](#IOError)                                                   	| Error in IO operation.                                 	| i_o_error                                                    |
+| [`RuntimeExceptionInLoaderError`](#RuntimeExceptionInLoaderError)       	| RuntimeException while loading GTFS dataset in memory. 	| runtime_exception_in_loader_error                            |
+| [`RuntimeExceptionInValidatorError`](#RuntimeExceptionInValidatorError) 	| RuntimeException while validating GTFS archive.        	| runtime_exception_in_validator_error                         |
+| [`ThreadExecutionError`](#ThreadExecutionError)                         	| ExecutionException during multithreaded validation     	| thread_execution_error                                       |
+| [`URISyntaxError`](#URISyntaxError)                                     	| A string could not be parsed as a URI reference.       	| u_r_i_syntax_error                                           |
 
-## Notices
+## Errors
 
-### Errors
+<a name="BlockTripsWithOverlappingStopTimesNotice"/>
```

Comment:
> I believe these anchors for the notice class name that are identical to the Markdown header are redundant

That's true, I removed them in 2a0345600b3e055001d0a822deec7139062a17e9.

For the Class names VS code, I agree with you, this isn't great. Here is how we quickly went about it with @maximearmstrong:
- option 1: replace NoticeClassName by the notice_code in the documentation
  - pros: clear, only one thing to reference a notice
  - cons: 
    - the existing documentation is using NoticeClassName and [some people](https://dot.ca.gov/cal-itp/Critical-GTFS-Validation-Errors#:~:text=has%20deemed%20critical.-,GTFS%20Schedule,-The%20following%20errors) refer to the notices with it.
    - the people looking at RULES.md want to find the description for: (1) the notice, and (2) the notice schema field names. Maybe it's just me, but when I started using the validator, `invalid_phone_number` and `fileName` was the same "thing": a variable that I need to find the description for, so I didn't understand why these had 2 different formats. So from a readability POV it would make sense to have these two things the same format.
- option 2: deprecate notice_code and use NoticeClassName to refer to the notices, and update the JSON report with new code
  - cons: users are currently relying on this code when this validator is integrated in a pipeline, this is a major change.
- option 3 (what we decided to go for right now): keep both for now: add the NoticeClassName in the HTML output and the notice_code in the table at the top of RULES.md
  - pros: users don't have to "guess" which maps to which
  - con: users are confused: why do we have this redundant info? what is notice_code for? Which I tried to mitigate with adding _Notice Code (as it appears in the report.json file)_ in the table 🙈

Does this seem like a reasonable tradeoff? 
This probably deserves more discussion. The most important thing to solve right now is:
- users going back-and-forth between the 2 documents
- users not having the same info in the report VS the documentation

Reply:
Okay, after reading your comment & doing a a quick research on how existing users are referencing the notices, I'm also now leaning more towards option 1. I think there is more pro (new users having a better experience) than cons (some existing users might do minor adjustments). @maximearmstrong, if you're OK with this, I will modify this PR to remove class names altogether. 
You would also need to update the HTML output to reflect the change.

Reply:
All good for me @isabelle-dr, I'm in favor of option 1 as well.